cli/src/git_util.rs

```diff
@@ -155,16 +156,69 @@ pub fn with_remote_git_callbacks<T>(
     callbacks.progress = progress_callback
         .as_mut()
         .map(|x| x as &mut dyn FnMut(&git::Progress));
+    // Based on Git's implementation: https://github.com/git/git/blob/43072b4ca132437f21975ac6acc6b72dc22fd398/sideband.c#L178
     let mut sideband_progress_callback = None;
+    let mut scratch = Vec::new();
+    let display_prefix = "remote: ";
+    let suffix = if std::io::stdout().is_terminal() {
+        "\x1B[K"
+    } else {
+        "        "
+    };
```

Comment:
nit: All of these variables are used only inside `if print_sideband_messages`, so move them into that block

Reply:
I shifted the other variables into the block, but I didn't move `scratch` since it would have been dropped at the end of the block. I think the correct fix is to shift it into the block and make the closure `move` so that `scratch` would be moved into the closure, but that causes a move of `ui` as well. Should I do that? (My understanding is I can wrap `ui` with an `Arc` to allow cloning and moving it into the closure.)

Reply:
> wrap `ui` with an `Arc` to allow cloning and moving it into the closure.

Perhaps, we can remove `&mut` requirements from various prompt functions.

Reply:
Thanks for the tip, I've updated this and update the closure to be `move` as well.